tools/shared.py

```diff
@@ -2730,16 +2730,21 @@ def clang_preprocess(filename):
   return run_process([CLANG_CC, '-DFETCH_DEBUG=1', '-E', '-P', '-C', '-x', 'c', filename], check=True, stdout=subprocess.PIPE).stdout
 
 def read_and_preprocess(filename):
-  f = open(filename, 'r').read()
-  pos = 0
-  include_pattern = re.compile('^#include\s*["<](.*)[">]\s?$', re.MULTILINE)
-  while(1):
-    m = include_pattern.search(f, pos)
-    if not m:
-      return f
-    included_file = open(os.path.join(os.path.dirname(filename), m.groups(0)[0]), 'r').read()
+  temp_dir = configuration.get_temp_files().get_dir()
+
+  # Create a settings file with the current settings to pass to the JS preprocessor
+  settings_str = "var " + ";\nvar ".join(Settings.serialize()[1::2])
```

Comment:
please explain in a comment  why we need `1::2` here. (I believe it is to skip the `-s` stuff from `-s K1=V1 -s K2=V2` etc.?)